static/js/dialog_widget.js

```diff
@@ -119,14 +119,14 @@ export function launch(conf) {
     }
 
     const submit_button = dialog.find("#dialog_submit_button");
-    submit_button.on("click", () => {
+    submit_button.on("click", (e) => {
         if (conf.loading_spinner) {
             show_dialog_spinner();
         } else if (conf.close_on_submit) {
             close_modal();
         }
         $("#dialog_error").empty();
-        conf.on_click();
+        conf.on_click(e);
```

Comment:
This would be great to make a preparatory commit that is just preserved when switching implementations.